internal/characters/cyno/skill.go

```diff
@@ -0,0 +1,182 @@
+
+	return action.ActionInfo{
+		Frames:          frames.NewAbilFunc(skillFrames),
+		AnimationLength: skillFrames[action.InvalidAction],
+		CanQueueAfter:   skillFrames[action.ActionDash], // earliest cancel
+		State:           action.SkillState,
+	}
+}
+
+func (c *char) skillB() action.ActionInfo {
+	c.tryBurstPPSlide(skillBHitmark)
+
+	ai := combat.AttackInfo{
+		ActorIndex:         c.Index,
+		Abil:               skillBName,
+		AttackTag:          combat.AttackTagElementalArt,
+		ICDTag:             combat.ICDTagNone,
+		ICDGroup:           combat.ICDGroupDefault,
+		Element:            attributes.Electro,
+		Durability:         25,
+		Mult:               skillB[c.TalentLvlSkill()],
+		HitlagFactor:       0.01,
+		HitlagHaltFrames:   0.03 * 60,
+		CanBeDefenseHalted: false,
+	}
+
+	if !c.StatusIsActive(a1Key) { // check for endseer buff
+		c.Core.QueueAttack(
+			ai,
+			combat.NewCircleHit(c.Core.Combat.Player(), 3, false, combat.TargettableEnemy),
+			skillBHitmark,
+			skillBHitmark,
+		)
+	} else {
+		// apply the extra damage on skill
+		c.judiscation()
+		if c.Base.Cons >= 1 && c.StatusIsActive(c1Key) {
+			c.c1()
+		}
+		if c.Base.Cons >= 6 { // constellation 6 giving 4 stacks on judication
+			c.c6stacks += 4
+			c.AddStatus("cyno-c6", 480, true) // 8s*60
+			if c.c6stacks > 8 {
+				c.c6stacks = 8
+			}
+		}
+
+		c.Core.QueueAttack(
+			ai,
+			combat.NewCircleHit(c.Core.Combat.Player(), 3, false, combat.TargettableEnemy),
+			skillBHitmark,
+			skillBHitmark,
+		)
+		// Apply the extra hit
+		ai.Abil = "Duststalker Bolt"
+		ai.Mult = 1.0
+		ai.FlatDmg = c.Stat(attributes.EM) * 2.5 // this is the A4
+		ai.ICDTag = combat.ICDTagCynoBolt
+		ai.ICDGroup = combat.ICDGroupCynoBolt
+		ai.IsDeployable = true
+
+		// 3 instances
+		// TODO: timing (frames) of each instance
+		for i := 0; i < 3; i++ {
+			c.Core.QueueAttack(
+				ai,
+				combat.NewCircleHit(c.Core.Combat.Player(), 1, false, combat.TargettableEnemy),
+				skillBHitmark,
+				skillBHitmark,
+			)
+		}
+
+	}
+	if c.burstExtension < 2 { // burst can only be extended 2 times per burst cycle (up to 18s, 10s base and +4 each time)
+		c.ExtendStatus(burstKey, 240) // 4s*60
+		c.burstExtension++
+	}
+
+	var count float64 = 1 // 33% of generating 2 on furry form
+	if c.Core.Rand.Float64() < .33 {
+		count++
+	}
+	c.Core.QueueParticle("cyno", count, attributes.Electro, skillBHitmark+c.ParticleDelay)
+
+	c.SetCDWithDelay(action.ActionSkill, 180, 26)
+
+	f := skillBFrames
+	if c.StatusIsActive(a1Key) {
+		f = skillA1Frames
+	}
+	return action.ActionInfo{
+		Frames:          frames.NewAbilFunc(f),
+		AnimationLength: skillBFrames[action.InvalidAction],
+		CanQueueAfter:   skillBFrames[action.ActionJump], // earliest cancel is 3f before SkillBHitmark
+		State:           action.SkillState,
+	}
+}
+
+// TODO:I am applying the skill dmg bonus this way to ensure that the skill gets the dmg bonus even if endseer expires mid cast (this may not be neccesary)
+func (c *char) judiscation() {
```

Comment:
This is kind of a hacky way to implement the a1 dmg buff. Perhaps there is a better method?